### PR TITLE
Add the Animation Component

### PR DESCRIPTION
**Pull request by vitoriaHeliane**
#### What is the purpose of this pull request?
Add the Animation Component with drawer animation type.

#### How should this be manually tested?
The Mini Cart's sidebar is animated by this component. Check it out!
[Access the workspace.](https://waza--storecomponents.myvtex.com/)

#### Types of changes
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [X] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/Animation/index.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      {this.props.children}
+    </div>
+  )
+
+  render() {
+    const { isActive, from } = this.props
+    const style = ANIMATIONS[from]
+
+    return (
+      <Transition
```
Comment (ClaudivanFilho) [inline react/components/Animation/index.js]: put in one line

File: react/components/Animation/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+<Animation> 
+  This is an animated text! \o/
+</Animation>
+```
+
+| Prop name          | Type       | Description                                                                 |
+| ------------------ | ---------- | --------------------------------------------------------------------------- |
+| `className`        | `String!`  | The className to set the animation                                          |
+| `children`         | `Node!`    | Component children that will be displayed inside of the animation           |
+| `from`             | `String!`  | Set the origin of the animation                                             |
```
Comment (ClaudivanFilho) [inline react/components/Animation/README.md]: put here the values that `from` can be ('left', 'right', ...)

**Comment by lucasecdb** [comment]
@vitoriaHeliane I think the component name is misleading, `Animation` is a much more broad concept and this is just a drawer-like animation. Maybe it'd be better to rename it to something among the lines of that.

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+- `AnimationComponent`.
```
Comment (lucasecdb) [inline CHANGELOG.md]: > `Animation` component.

File: react/components/Animation/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+You can use it in your code like a React component with the jsx tag: `<Animation />`. 
+```html
+<Animation> 
+  This is an animated text! \o/
+</Animation>
+```
+
+| Prop name          | Type       | Description                                                                 |
+| ------------------ | ---------- | --------------------------------------------------------------------------- |
+| `className`        | `String!`  | The className to set the animation                                          |
```
Comment (ClaudivanFilho) [inline react/components/Animation/README.md]: `!` indicates that the field is required :warning:

File: react/components/Animation/index.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+
+import { Transition } from 'react-spring'
+
+const ANIMATIONS = {
+  'left': {
```
Comment (lucasecdb) [inline react/components/Animation/index.js]: You don't need the quotes in the keys of this object.

File: react/components/Animation/index.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  static defaultProps = {
+    className: '',
+    from: 'left',
+  }
+
+  renderChildren = style => (
+    <div className={this.props.className}
+      style={style}>
```
Comment (lucasecdb) [inline react/components/Animation/index.js]: Put the `>` in a new line.

File: react/components/DrawerAnimation/README.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+DrawerAnimation is a component that any VTEX app can import.
+
+To import it into your code: 
+```js
+import DrawerAnimation from 'vtex.store-components/DrawerAnimation'
+```
+
+## Usage
+You can use it in your code like a React component with the jsx tag: `<DrawerAnimation />`. 
+```html
```
Comment (lucasecdb) [inline react/components/DrawerAnimation/README.md]: Use `jsx`, not `html`

File: CHANGELOG.md [hunk 8]
```diff
@@ -6,6 +6,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+- `Animation` component.
```
Comment (ClaudivanFilho) [inline CHANGELOG.md]: `DrawerAnimation`

File: react/components/Animation/index.js [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+    isActive: PropTypes.bool,
+    /* Classname to the animation */
+    className: PropTypes.string,
+    /* Type of animation */
+    type: PropTypes.oneOf(['drawerLeft', 'drawerRight', 'drawerTop', 'drawerBottom']),
+  }
+
+  static defaultProps = {
+    className: '',
+    from: 'drawerLeft',
```
Comment (ClaudivanFilho) [inline react/components/Animation/index.js]: type

**Comment by lucasecdb** [comment]
[Add the Animation Component](https://trello.com/c/YdXTJN13/55-add-the-animation-component)

**Pull request merged**